internal/pkg/specs/specs.go

```diff
@@ -65,3 +71,47 @@ func (projects *ProjectsSpec) Contains(projectSpec *ProjectSpec) bool {
 	}
 	return false
 }
+
+// RepoFromPath returns a ProjectSpec from a repo at the given path
+func RepoFromPath(path string) (ProjectSpec, error) {
+	repo, err := git.PlainOpenWithOptions(".", &git.PlainOpenOptions{DetectDotGit: true})
```

Comment:
Should this be using the path instead of hardcoding `"."`?

```suggestion
	repo, err := git.PlainOpenWithOptions(path, &git.PlainOpenOptions{DetectDotGit: true})
```

Reply:
Shoot yeah